firelink/apps.py

```diff
@@ -51,6 +51,43 @@ def _process_bonfire_apps_list(self, apps):
         apps_array.sort(key=lambda app: app['name'])
         return apps_array
 
+    def get_processed_template(self, request):
+        """Get the processed template for an app."""
+        namespace = request["namespace"]
+        clowd_env = self._get_clowdenv_for_ns(namespace)
+        if not clowd_env:
+            raise bonfire.FatalError("Could not find a ClowdEnvironment tied to ns '{ns}'.")
+        request["clowd_env"] = clowd_env
+        apps_config = bonfire._process(
+            request["app_names"],
+            request["source"],
+            request["get_dependencies"],
+            request["optional_deps_method"],
+            request["local_config_method"],
+            request["set_image_tag"],
+            request["ref_env"],
+            request["fallback_ref_env"],
+            request["target_env"],
+            request["set_template_ref"],
+            request["set_parameter"],
+            request["clowd_env"],
+            request["local_config_path"],
+            AppOrComponentSelector(True, request["remove_resources"], []),
+            AppOrComponentSelector(False, request["no_remove_resources"], []),
+            AppOrComponentSelector(False, [], request["remove_dependencies"]),
+            AppOrComponentSelector(True, [], request["no_remove_dependencies"]),
```

Comment:
I am not as familiar with the options that the user has in the UI that controls how `remove_resources/no_remove_resources/remove_dependencies/no_remove_dependencies` get set. Just want to double check the intention here...

If the first arg is `True` this is the equivalent of passing in `--remove-resources all`. Is that the intent? Same with `--no-remove-dependencies`

The logic for how this is "translated" from the CLI commands into `AppOrComponentSelector` is controlled here: https://github.com/RedHatInsights/bonfire/blob/master/bonfire/bonfire.py#L367-L430 and it is admittedly a little wonky because it is cross-checking `click` parameter values. Let's chat about what the desired outcome is here...

Reply:
I sent up a revision that I _think_ addresses what you were saying.